plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiMetadata.java

```diff
@@ -162,12 +169,30 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
     {
         HudiTableHandle handle = (HudiTableHandle) tableHandle;
         HudiPredicates predicates = HudiPredicates.from(constraint.getSummary());
+        TupleDomain<HiveColumnHandle> regularColumnPredicates = predicates.getRegularColumnPredicates();
+        TupleDomain<HiveColumnHandle> partitionColumnPredicates = predicates.getPartitionColumnPredicates();
+
+        // TODO Since the constraint#predicate isn't utilized during split generation. So,
+        //  Let's not add constraint#predicateColumns to newConstraintColumns.
+        Set<HiveColumnHandle> newConstraintColumns = Stream.concat(
```

Comment:
Regarding

> `Constraint#predicate` is not getting utilized during split generation that means all the data files will be read for such cases where predicate does not get translated to `TupleDomain`.

It could have been a miss in the first implementation, probably due to lack of column stats index. What do we lose if we include `Constraint#predicate` in `newConstraintColumns`? I think we should still include it in the metadata layer. Using it in the split manager layer can be fixed in another PR.

Reply:
> What do we lose if we include Constraint#predicate in newConstraintColumns? 

Thanks @codope for the review. Since during spilt generation we do not filter out the data files based on the `Constraint#predicate` then even if `query-partition-filter-required` is enforced all the data files will be read, which is the false alarm.

I have mentioned the `TODO` comment in the `HudiMetadata` class. Once Hudi uses `Constraint#predicate` then we can include `Constraint#predicate` in newConstraintColumns.